iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -580,6 +580,14 @@ def align_index(self, constraints: list["Constraint"]) -> None:
         """
         pass
 
+    def transform_index_backwards(
```

Comment:
Can you combine this with transform_index and add arg as an optional argument? They seem similar in spirit which is that they both modify the index (for permute it comes from the semantics, and for read/write from the dynamic vals).

Reply:
Actually, that's something I didn't quite understand - the difference between forward and backward. What is a forward transformation and what is a backward transformation? Aren't they all just transformations of the index sequence?

Reply:
So based on our discussion, the forward and backward transformations relate to apply a function f() and its inverse f_inv(). Could you add a comment describing this ?